File: Find_Players_With_Zero_or_One_Losses.js
/**
 * @param {number[][]} matches
 * @return {number[][]}
 */
var findWinners = function(matches) {
    const res = [[], []];
    const loseTimes = {};

    for (const match of matches) {
        const [w, l] = match;
        if (!(w in loseTimes)) {
            loseTimes[w] = 0;
        }
        loseTimes[l] = 1 + (loseTimes[l] || 0);
    }

    for (const [player, loseCount] of Object.entries(loseTimes)) {
        if (loseCount < 2) {
            res[loseCount].push(parseInt(player));
        }
    }

    return [res[0].sort((a, b) => a - b), res[1].sort((a, b) => a - b)];    
};

File: Find_Players_With_Zero_or_One_Losses.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        res = [[] for _ in range(2)]
        lose_times = {}

        for w, l in matches:
            if w not in lose_times:
                lose_times[w] = 0
            lose_times[l] = 1 + lose_times.get(l, 0)
        
        for player, lose_count in lose_times.items():
            if lose_count < 2:
                res[lose_count].append(player)
        
        return [sorted(res[0]), sorted(res[1])]
